### ISSUE TITLE
Dropdown header not accounted for in placement

### ISSUE THREAD

**Issue by patrickjtoy** [themesberg/flowbite-react#972]
- [x] I have searched the [Issues](https://github.com/themesberg/flowbite-react/issues) to see if this bug has already been reported
- [x] I have tested the latest version

## Steps to reproduce

1. Add a `Dropdown.Header` to a `Dropdown` component.
2. On a screen where auto placement needs to be calculated, open the dropdown.
3. Note that the main `Dropdown.Item` list is correctly calculated for, however the `Dropdown.Header` is not.

## Current behavior

The `Dropdown.Header` component is being rendered outside the visible area.

## Expected behavior

The entire `Dropdown` component should be controlled by the FloatingUI placement, ensuring the whole component is visible.

## Context

![image](https://github.com/themesberg/flowbite-react/assets/5121391/e9260058-a829-450d-9d08-35c67a9b9c74)

**Comment by rluders** [comment]
@patrickjtoy thanks for reporting it. Are u using the `0.6.0`?

**Comment by patrickjtoy** [comment]
> @patrickjtoy thanks for reporting it. Are u using the `0.6.0`?

Yes I am. I also noticed that this issue appears on an iPad Pro screen size, but not for larger screens, if that helps your troubleshooting!
